sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -376,6 +400,170 @@ private static MessagingEntityType validateEntityPaths(ClientLogger logger, Stri
         return entityType;
     }
 
+
+    @ServiceClientBuilder(serviceClients = {ServiceBusSessionReceiverAsyncClient.class})
+    public final class ServiceBusSessionReceiverAsyncClientBuilder {
+        private String topicName;
+        private String queueName;
+        private String subscriptionName;
+        private ReceiveMode receiveMode;
+        private String sessionId;
+        ServiceBusSessionReceiverAsyncClientBuilder () {
+
+        }
+        /**
+         * Return instance of {@link ServiceBusMultiSessionProcessorClientBuilder} to receive messages from multiple sessions.
+         * @return {@link ServiceBusSessionReceiverAsyncClient} instance.
+         */
+        public ServiceBusSessionReceiverAsyncClient buildAsyncClient() {
+            return null;
+        }
+        public ServiceBusSessionReceiverAsyncClientBuilder receiveMode(ReceiveMode receiveMode) {
+            this.receiveMode = receiveMode;
+            return this;
+        }
+        public ServiceBusSessionReceiverAsyncClientBuilder sessionId(String sessionId) {
+            this.sessionId = sessionId;
+            return this;
+        }
+        /**
+         * Sets the name of the subscription in the topic to listen to.
+         *
+         * @param subscriptionName Name of the subscription.
+         *
+         * @return The modified {@link ServiceBusMultiSessionProcessorClientBuilder} object.
+         * @see #topicName A topic name should be set as well.
+         */
+        public ServiceBusSessionReceiverAsyncClientBuilder subscriptionName(String subscriptionName) {
+            this.subscriptionName = subscriptionName;
+            return this;
+        }
+
+        /**
+         * Sets the name of the topic.
+         *
+         * @param queueName Name of the queue.
+         *
+         * @return The modified {@link ServiceBusMultiSessionProcessorClientBuilder} object.
+         * @see #subscriptionName A subscription name should be set as well.
+         */
+        public ServiceBusSessionReceiverAsyncClientBuilder queueName(String queueName) {
+            this.queueName = queueName;
+            return this;
+        }
+
+        /**
+         * Sets the name of the topic.
+         *
+         * @param topicName Name of the topic.
+         *
+         * @return The modified {@link ServiceBusMultiSessionProcessorClientBuilder} object.
+         * @see #subscriptionName A subscription name should be set as well.
+         */
+        public ServiceBusSessionReceiverAsyncClientBuilder topicName(String topicName) {
```

Comment:
I think I've asked you this before but can you not just have a builder method that takes (topicName, subscriptionName) so you don't ever omit one of them on accident?